File: cpp-examples/classes/advanced/BookShelf.cpp
#include "Book.h"
#include <iostream>
using std::cout;
using std::endl;


int main() {
    
    Book book1("An Introduction to Dynamic Programming", "Timothy Budd", 1991);
    book1.printDetails();
    // Oops wrong title! Let's update it!
    book1.updateTitle("An Introduction to Object-Oriented Programming");
    book1.printDetails();
    Book book2("AI Literacy Fundamentals", "Ben Jones");
    book2.printDetails();
    cout << "Total Books: " << Book::getBookCount() << endl;
    book1.printAuthor();
    return 0;
}

File: python-examples/classes/advanced/BookShelf.py
from Book import Book

def main():
    book1 = Book("An Introduction to Dynamic Programming", "Timothy Budd", 1991)
    book1.print_details()
    book1.update_title("An Introduction to Object-Oriented Programming")
    book1.print_details()
    book2 = Book("AI Literacy Fundamentals", "Ben Jones")
    book2.print_details()
    print(f"Total Books: {Book.get_book_count()}")
    Book.static_method_example()


if __name__ == "__main__":
    main()